### DISCUSSION TITLE
Some question about functionalities ?

### DISCUSSION THREAD

**Discussion by reyman** [rasendubi/uniorg#86] [Q&A] [answered]
Hi, 

I'm happy to found an org parser that publish directly to html with astro !

* About literate programming 

I'm using a lot org-mode with org-babel and noweb for literate programming.

 When you say this is **work-in-progress** in the Readme, what happen during parsing of such Literate Programming files ? 

* About tag and filtering org-roam notes to output different website using route

Do you think it will be possible to define different path/website from a single org-roam base, for example using : 
- tag ?  I develop a little the problem [here with schema](https://github.com/org-roam/org-roam/issues/911) and here  : https://github.com/emacs-love/weblorg/issues/18  
- dynamic block ? Like here : https://github.com/org-roam/org-roam/issues/2251 

* About org-transclusion ?

Do you think uniorg parser works with the fabulous [org-transclusion](https://github.com/nobiot/org-transclusion) package ? 

Thanks !

**Comment by rasendubi** [comment, marked as answer]
Hey

**Literate programming**: org-babel parsing works. It will export the source code. You can test this in [playground](https://oleksii.shmalko.com/uniorg/).

**Tags and filtering**: that should be possible. `org-astro` allows you to import org files as Astro components (and access their metadata) and it's up to you how you filter and map it to paths. (As an example, [here](https://github.com/rasendubi/oleksii.shmalko.com/blob/5d5dc3e17a9994a2fec94d17cfcd1fb1352e9e4c/src/lib/posts.ts#L24-L35) is a filtering by directory and `#+DRAFT:` keyword.)

**dynamic blocks**: dynamic blocks parsing currently is not implemented (it is parsed as `begin` + content + `end` keywords). For exporting this basically means that content is exported as is. So as long as your org file has up-to-date block value, it should be exported correctly.

If there is interest in parsing dynamic blocks properly (e.g., to update their value from js), I can guide on how to implement this (or can implement it when I have more time).

**org-transclusion**: org-transclusion is not supported (and likely won't be supported in the core uniorg), but it can be implemented as a plugin.
